cluster-operator/src/main/java/io/strimzi/operator/cluster/model/Utils.java

```diff
@@ -0,0 +1,127 @@
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class Utils {
+    public static int getInteger(Map<String, String> data, String key, int defaultValue) {
+        try {
+            if (data.get(key) == null) { // value is not set in config map -> will be used default value
+                return defaultValue;
+            }
+            return Integer.parseInt(data.get(key));
+        } catch (NumberFormatException e)  {
+            String msg = " is corrupted";
+            if (data.get(key).isEmpty()) {
+                msg = " is empty";
+            }
+            throw new InvalidConfigMapException(key, msg);
+        }
+    }
+
+    public static boolean getBoolean(Map<String, String> data, String key, boolean defaultValue) {
+        if (data.get(key) == null) { // value is not set in config map
+            return defaultValue;
+        } // java parses "truue" as false. We need to be more strict
+        if (data.get(key).equals("true")) {
+            return true;
+        } else if (data.get(key).equals("false")) {
+            return false;
+        } else {
+            throw new InvalidConfigMapException(key, " is corrupted");
+        }
+    }
+
+    public static Storage getStorage(Map<String, String> data, String key) {
+        try {
+            if (data.get(key) == null) { // value is not set in config map
+                return new Storage(Storage.StorageType.PERSISTENT_CLAIM);
+            }
+            return Storage.fromJson(new JsonObject(data.get(key)));
+        } catch (Exception e) {
+            throw new InvalidConfigMapException(key, " is corrupted");
+        }
+    }
+
+    public static String getNonemptyString(Map<String, String> data, String key, String defaultValue) {
+        if (data.get(key) == null) { // value is not set in config map -> will be used default value
+            return defaultValue;
+        }
+        if (data.get(key).length() == 0) {
+            throw new InvalidConfigMapException(key, " is empty");
+        }
+        return data.get(key);
+    }
+
+    public static String getJsonCorruptionBlame(DecodeException de, String config) {
+        String token = "";
+        token = de.getMessage();
+        if (token.contains("Illegal unquoted character")) {
+            return "JSON quotation";
+        }
+        token = token.replace("Failed to decode: Unrecognized token '", "");
+        token = token.substring(0, token.indexOf("'"));
+
+        String[] entries = config.replace("{", "").replace("}", "")
+               .replace("\"", "").replace(" ", "")
+               .replace("\n", "").split(",");
+        int i;
+        String blame = "";
+        for (i = 0; i < entries.length; ++i) {
+            if (entries[i].split(":")[1].equals(token)) {
+                blame = entries[i].split(":")[0];
+                break;
+            }
+        }
+        return blame;
+    }
+
+    public static JsonObject getConfig(Map<String, String> data, String key) {
```

Comment:
Leaving other comments aside ... I have major doubts about this method. It will not work as designed now. You are trying to validate too much in the same method. For example:

The `kafka-config`, `connect-config` and `zookeeper-config` fields have completely different validation requirements than the `kafka-metrics-config` and `zookeeper-metrics-config` fields.

The config fields are just key-value maps. But the metrics are a real JSON objects. So for example the metrics config cannot be validated by this method because it can contain another JSON objects or arrays. When validating the regular configs, this method can work. But it basically duplicates the functionality in `AbstractConfiguration` with the only difference that it throws exceptions instead of ignoring the invalid fields with error message.

My two cents are that we should:
* Have a generic method which validates that the field is JsonObject
* Change the `AbstractConfiguration` to throw exceptions for invalid fields inside the JSON and use it in a method similar to `Utils.getStorage`

PS: Have you tried to run this with real cluster? I haven't but my guess is that it would fail with the example YAML files.